src/containers/Tenant/Diagnostics/Overview/TableInfo/TableInfo.tsx

```diff
@@ -0,0 +1,90 @@
+import React, {useMemo} from 'react';
+import cn from 'bem-cn-lite';
+
+import type {TEvDescribeSchemeResult} from '../../../../../types/api/schema';
+
+import {InfoViewer, InfoViewerItem} from '../../../../../components/InfoViewer';
+
+import {getEntityName} from '../../../utils';
+
+import {prepareTableInfo} from './prepareTableInfo';
+
+import i18n from './i18n';
+
+import './TableInfo.scss';
+
+const b = cn('ydb-diagnostics-table-info');
+
+const checkIsEmptyTableInfo = (
+    generalTableInfo: InfoViewerItem[],
+    tabletMetricsInfo: InfoViewerItem[],
+    partitionConfigInfo: InfoViewerItem[],
+    tableStatsInfo: InfoViewerItem[][],
+) => {
+    return (
+        [generalTableInfo, tabletMetricsInfo, partitionConfigInfo, tableStatsInfo.flat()].flat()
+            .length === 0
+    );
+};
+
+interface TableInfoProps {
+    data?: TEvDescribeSchemeResult;
+}
+
+export const TableInfo = ({data}: TableInfoProps) => {
+    const entityName = getEntityName(data?.PathDescription);
+
+    const {
+        generalTableInfo = [],
+        tableStatsInfo = [],
+        tabletMetricsInfo = [],
+        partitionConfigInfo = [],
+    } = useMemo(() => prepareTableInfo(data), [data]);
```

Comment:
Is `useMemo` really needed here?

Reply:
i'd say yes. `prepareTableInfo` is a complex function